### ISSUE TITLE
GHC 9.4.6 errors.

### ISSUE THREAD

**Issue by julialongtin** [minoki/haskell-floating-point#4]
```
Building library for rounded-hw-0.3.0..
[ 1 of 24] Compiling Numeric.Rounded.Hardware.Internal.FloatUtil ( src/Numeric/Rounded/Hardware/Internal/FloatUtil.hs, dist/build/Numeric/Rounded/Hardware/Internal/FloatUtil.o, dist/build/Numeric/Rounded/Hardware/Internal/FloatUtil.dyn_o )
[ 2 of 24] Compiling Numeric.Rounded.Hardware.Internal.Rounding ( src/Numeric/Rounded/Hardware/Internal/Rounding.hs, dist/build/Numeric/Rounded/Hardware/Internal/Rounding.o, dist/build/Numeric/Rounded/Hardware/Internal/Rounding.dyn_o )
[ 3 of 24] Compiling Numeric.Rounded.Hardware.Internal.Conversion ( src/Numeric/Rounded/Hardware/Internal/Conversion.hs, dist/build/Numeric/Rounded/Hardware/Internal/Conversion.o, dist/build/Numeric/Rounded/Hardware/Internal/Conversion.dyn_o )
[ 4 of 24] Compiling Numeric.Rounded.Hardware.Internal.Constants ( src/Numeric/Rounded/Hardware/Internal/Constants.hs, dist/build/Numeric/Rounded/Hardware/Internal/Constants.o, dist/build/Numeric/Rounded/Hardware/Internal/Constants.dyn_o )
[ 5 of 24] Compiling Numeric.Rounded.Hardware.Internal.Class ( src/Numeric/Rounded/Hardware/Internal/Class.hs, dist/build/Numeric/Rounded/Hardware/Internal/Class.o, dist/build/Numeric/Rounded/Hardware/Internal/Class.dyn_o )
[ 6 of 24] Compiling Numeric.Rounded.Hardware.Backend.X87LongDouble ( src/Numeric/Rounded/Hardware/Backend/X87LongDouble.hs, dist/build/Numeric/Rounded/Hardware/Backend/X87LongDouble.o, dist/build/Numeric/Rounded/Hardware/Backend/X87LongDouble.dyn_o )
[ 7 of 24] Compiling Numeric.Rounded.Hardware.Backend.ViaRational ( src/Numeric/Rounded/Hardware/Backend/ViaRational.hs, dist/build/Numeric/Rounded/Hardware/Backend/ViaRational.o, dist/build/Numeric/Rounded/Hardware/Backend/ViaRational.dyn_o )
[ 8 of 24] Compiling FFIWrapper.Float ( src/FFIWrapper/Float.hs, dist/build/FFIWrapper/Float.o, dist/build/FFIWrapper/Float.dyn_o )
[ 9 of 24] Compiling FFIWrapper.Double ( src/FFIWrapper/Double.hs, dist/build/FFIWrapper/Double.o, dist/build/FFIWrapper/Double.dyn_o )
[10 of 24] Compiling Numeric.Rounded.Hardware.Backend.C ( src/Numeric/Rounded/Hardware/Backend/C.hs, dist/build/Numeric/Rounded/Hardware/Backend/C.o, dist/build/Numeric/Rounded/Hardware/Backend/C.dyn_o )

src/Numeric/Rounded/Hardware/Backend/C.hs:94:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘intervalFloatFromInteger’
   |
94 | intervalFloatFromInteger x
   | ^^^^^^^^^^^^^^^^^^^^^^^^
[11 of 24] Compiling Numeric.Rounded.Hardware.Backend.FastFFI ( src/Numeric/Rounded/Hardware/Backend/FastFFI.hs, dist/build/Numeric/Rounded/Hardware/Backend/FastFFI.o, dist/build/Numeric/Rounded/Hardware/Backend/FastFFI.dyn_o )

src/Numeric/Rounded/Hardware/Backend/FastFFI.hs:219:62: error:
    • Couldn't match type ‘Int64#’ with ‘Int#’
      Expected: INT64#
        Actual: Int64#
    • In the first argument of ‘fastIntervalFromInt64#’, namely ‘x’
      In the expression: fastIntervalFromInt64# x
      In the expression:
        case fastIntervalFromInt64# x of (# l, h #) -> (D# l, D# h)
    |
219 | fastIntervalFromInt64 (I64# x) = case fastIntervalFromInt64# x of
    |                                                              ^
[13 of 24] Compiling Numeric.Rounded.Hardware.Internal.Show ( src/Numeric/Rounded/Hardware/Internal/Show.hs, dist/build/Numeric/Rounded/Hardware/Internal/Show.o, dist/build/Numeric/Rounded/Hardware/Internal/Show.dyn_o )
Error: cabal: Failed to build rounded-hw-0.3.0 (which is required by
test:test-stat from hslice-0.0.1, test:test-hslice from hslice-0.0.1 and
others). See the build log above for details.

```

**Comment by minoki** [comment]
Isn't this one already fixed by [PR #2](https://github.com/minoki/haskell-floating-point/pull/2)?

**Comment by julialongtin** [comment]
No, a pull of three day ago's master failed with the above.

Tested on debian unstable, running GHC from the package ghc 9.4.6-1.

It's worth noting that CI seems to have building rounded-hw disabled.

**Comment by minoki** [comment]
Are you using rounded-hw as a dependency from some other package called `hslice`? Then you need to add `source-repository-package` to `cabal.project` to use the unreleased version:

```
source-repository-package
    type: git
    location: https://github.com/minoki/haskell-floating-point.git
    tag: c07327376316b5a0ba852451b10d0d6047a1deb4
    subdir: rounded-hw
```

As for CI, GHC 9.4 was disabled because of [a GHC bug](https://gitlab.haskell.org/ghc/ghc/-/issues/23762). It's re-enabled in `develop` branch.

I understand that I need to make a new release, but there's a trouble with GHC 9.8, so I cannot assure when I can make the release.

**Comment by minoki** [comment]
I've released rounded-hw v0.4.0 and I believe it is compatible with GHC 9.4. Please try it out!

**Comment by julialongtin** [comment]
works for me!
